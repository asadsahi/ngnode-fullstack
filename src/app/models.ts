import { InjectionToken, TemplateRef } from '@angular/core';
import { ValidatorFn } from '@angular/forms';

export const COOKIES = new InjectionToken<string>('COOKIES');
export interface KeyValuePair<T> {
  key: string;
  value: T;
}

export interface ISocialLogins {
  loginProvider: string;
  providerKey: string;
  providerDisplayName: string;
  active: boolean;
}

export interface ITwoFactorModel {
  hasAuthenticator: boolean;

  recoveryCodesLeft: number;

  is2faEnabled: boolean;
}

export interface IEnableAuthenticatorModel {
  code: string;
  sharedKey: string;
  authenticatorUri: string;
}

export interface IModalOptions {
  title: string;
  message?: string;
  template?: any;
}
export enum FieldTypes {
  Textbox = 'input',
  FileUpload = 'file',
  Password = 'password',
  Email = 'email',
  Number = 'number',
  Date = 'date',
  Time = 'time',
  Textarea = 'textarea',
  Radiolist = 'radiolist',
  Select = 'select',
  Checkbox = 'checkbox',
  Checkboxlist = 'checkboxlist',
  Button = 'button',
}

export interface Field {
  config: IFieldConfig;
}
export interface IFieldConfig {
  name: string;
  label?: string;
  disabled?: boolean;
  options?: IOption[];
  placeholder?: string;
  type: FieldTypes;
  validation?: ValidatorFn[];
  value?: any;
  onSubmit?: Function;
  errorMessages?: Object;
}

export interface IOption {
  key: string | number;
  value: string;
  selected?: boolean;
}
