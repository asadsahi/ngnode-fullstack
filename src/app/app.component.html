<!-- app modal component -->
<appc-loading></appc-loading>

<app-toast aria-live="polite" aria-atomic="true"></app-toast>

<ng-template modal>
  <appc-modal-component></appc-modal-component>
</ng-template>

<!-- header component -->
<appc-header></appc-header>

<!-- component routing placeholder -->
<main role="main">
  <router-outlet #o="outlet"></router-outlet>
</main>

<!-- footer component -->
<appc-footer></appc-footer>
