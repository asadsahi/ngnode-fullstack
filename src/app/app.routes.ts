import { Routes } from '@angular/router';
import { HomeComponent } from '@app/home/home.component';
import { PrivacyComponent } from '@app/components';

export const routes: Routes = [
  { path: '', component: HomeComponent, pathMatch: 'full', data: { state: 'home' } },
  { path: 'examples', loadChildren: () => import('./+examples/examples.module').then(m => m.ExamplesModule) },
  { path: 'privacy', component: PrivacyComponent }
];
