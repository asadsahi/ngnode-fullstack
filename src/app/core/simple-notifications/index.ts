export * from './simple-notifications.module';
