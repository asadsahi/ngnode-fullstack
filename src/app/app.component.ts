import { Component, OnInit } from '@angular/core';
import { Title, Meta } from '@angular/platform-browser';
import { Router } from '@angular/router';

// import { routerTransition } from './router.animations';
import { AppService, AuthService } from '@app/services';
@Component({
  selector: 'appc-root',
  // animations: [routerTransition],
  styleUrls: ['./app.component.scss'],
  templateUrl: './app.component.html'
})
export class AppComponent implements OnInit {
  constructor(
    private authService: AuthService,
    private router: Router,
    private title: Title,
    private meta: Meta,
    private appService: AppService
  ) { }

  public ngOnInit() {
    this.updateTitleAndMeta();
    if (window.location.href.indexOf('?postLogout=true') > 0) {
      this.authService.signoutRedirectCallback().then(() => {
        const url: string = this.router.url.substring(
          0,
          this.router.url.indexOf('?')
        );
        this.router.navigateByUrl(url);
      });
    }
  }

  public getState(outlet: any) {
    return outlet.activatedRouteData.state;
  }

  private updateTitleAndMeta() {
    this.title.setTitle(this.appService.appData.content['app_title']);
    this.meta.addTags([
      { name: 'description', content: this.appService.appData.content['app_description'] },
      { property: 'og:title', content: this.appService.appData.content['app_title'] },
      { property: 'og:description', content: this.appService.appData.content['app_description'] }
    ]);
  }
}
