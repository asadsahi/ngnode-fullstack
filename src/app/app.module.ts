import { BrowserModule, BrowserTransferStateModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { RouterModule } from '@angular/router';
import { ServiceWorkerModule } from '@angular/service-worker';

// import { PrebootModule } from 'preboot';

import { environment } from '../environments/environment';

import { routes } from './app.routes';
// Components
import { FooterComponent, HeaderComponent, PrivacyComponent } from '@app/components';
import { AppComponent } from './app.component';
import { HomeComponent } from './home/home.component';
import { SharedModule } from './shared';

@NgModule({
  declarations: [
    // Components
    AppComponent,
    HomeComponent,
    FooterComponent,
    HeaderComponent,
    PrivacyComponent,
  ],
  imports: [
    BrowserModule.withServerTransition({ appId: 'ng-cli-universal' }),
    // PrebootModule.withConfig({ appRoot: 'appc-root' }),
    BrowserAnimationsModule,
    BrowserTransferStateModule,
    RouterModule.forRoot(routes, { initialNavigation: 'enabled' }),
    ServiceWorkerModule.register('ngsw-worker.js', { enabled: environment.production }),
    SharedModule,
  ],
  bootstrap: [AppComponent],
})
export class AppModule {}
