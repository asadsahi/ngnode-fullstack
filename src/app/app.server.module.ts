import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { ServerModule } from '@angular/platform-server';
import { AppComponent } from './app.component';
import { AppModule } from './app.module';
import { ModuleMapLoaderModule } from '@nguniversal/module-map-ngfactory-loader';
import { ServerTransferStateModule } from '@angular/platform-server';
import { AppShellComponent } from './app-shell/app-shell.component';

const routes: Routes = [{ path: 'shell', component: AppShellComponent }];

@NgModule({
    imports: [
        AppModule,
        NoopAnimationsModule,
        ServerTransferStateModule,
        ServerModule,
        ModuleMapLoaderModule,
        RouterModule.forRoot(routes),
    ],
    bootstrap: [AppComponent],
    providers: [
    ],
})
export class AppServerModule { }
