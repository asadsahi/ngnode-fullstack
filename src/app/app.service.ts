import { TransferState, makeStateKey } from '@angular/platform-browser';
import { Injectable, Inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';

const APP_DATA_KEY = makeStateKey<string>('appData');

@Injectable()
export class AppService {
    constructor(
        @Inject('ORIGIN_URL') private baseUrl: string,
        private transferState: TransferState,
        private http: HttpClient
    ) { }


    public get appData(): IApplicationConfig {
        return this.transferState.get(APP_DATA_KEY, null as IApplicationConfig);
    }
    getAppData(): Promise<IApplicationConfig> {
        const transferredAppData = this.transferState.get(APP_DATA_KEY, null as IApplicationConfig);
        if (!transferredAppData) {
            return this.http.get(`${this.baseUrl}/api/applicationdata`).toPromise()
                .then((data: IApplicationConfig) => {
                    this.transferState.set(APP_DATA_KEY, data);
                    return data;
                });
        }
        return new Promise((resolve, reject) => {
            resolve(transferredAppData);
        });
    }
}

export function getAppData(appService: AppService) {
    return () => appService.getAppData();
}
