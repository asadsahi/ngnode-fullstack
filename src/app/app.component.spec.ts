import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { TestBed, async, fakeAsync, tick } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from './app.component';
import { AppService } from '@app/app.service';
import { CoreModule } from '@app/core';
import { MockAppService } from '../../__mocks__/MockAppService';

describe('AppComponent', () => {
  let fixture;

  beforeEach(
    async(() => {
      TestBed.configureTestingModule({
        declarations: [AppComponent],
        imports: [
          CoreModule.forRoot(),
          NoopAnimationsModule,
          RouterTestingModule.withRoutes([])
        ],
        schemas: [NO_ERRORS_SCHEMA],
        providers: [
          { provide: AppService, useClass: MockAppService },
          { provide: 'ORIGIN_URL', useValue: 'http://mock.com' },
        ]
      }).compileComponents();
    })
  );

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
  });

  it('should be created without error',
    async(() => {
      const app = fixture.debugElement.componentInstance;
      expect(app).toBeTruthy();
    })
  );

  it('snapshot created successfully', () => {
    expect(fixture).toMatchSnapshot();
  });

});
