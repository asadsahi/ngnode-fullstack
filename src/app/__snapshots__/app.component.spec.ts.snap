// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent snapshot created successfully 1`] = `
<appc-root
  activatedRoute={[Function ActivatedRoute]}
  appService={[Function MockAppService]}
  baseUrl={[Function String]}
  meta={[Function Meta]}
  oauthService={[Function OAuthService]}
  platformId={[Function String]}
  router={[Function Router]}
  title={[Function Title]}
>
  <appc-header
    class="ng-tns-c3-0"
  >
    <appc-simple-notifications
      ng-reflect-options="[object Object]"
    >
      <div
        class="simple-notification-wrapper top right"
        ng-reflect-klass="simple-notification-wrapper"
        ng-reflect-ng-class="top,right"
      >
        
      </div>
    </appc-simple-notifications>
    <header
      class="header"
    >
      <nav
        class="inner"
      >
        <a
          href="/"
          ng-reflect-router-link=""
          routerlink=""
        >
          <img
            alt="logo"
            class="logo"
            src="/assets/android-chrome-192x192.png"
          />
        </a>
        <a
          href="/"
          ng-reflect-router-link=""
          ng-reflect-router-link-active="active"
          routerlink=""
          routerlinkactive="active"
        >
          app_nav_home
        </a>
        
        
        
        <a
          class="ng-star-inserted"
          href="/chat"
          ng-reflect-router-link="chat"
          ng-reflect-router-link-active="active"
          routerlink="chat"
          routerlinkactive="active"
        >
          app_nav_chat
        </a>
        
        <a
          class="ng-star-inserted"
          href="/register"
          ng-reflect-router-link="register"
          ng-reflect-router-link-active="active"
          routerlink="register"
          routerlinkactive="active"
        >
          app_nav_register
        </a>
        
        <a
          class="ng-star-inserted"
          href="/login"
          ng-reflect-router-link="login"
          ng-reflect-router-link-active="active"
          routerlink="login"
          routerlinkactive="active"
        >
          app_nav_login
        </a>
        
        <form
          action="/api/setlanguage"
          class="culture-form ng-star-inserted"
          method="post"
          ngnoform=""
        >
          <input
            id=""
            name="culture"
            type="submit"
            value=""
          />
        </form>
        <a
          class="github"
          href="app_repo_url"
          rel="noopener"
          target="_blank"
        >
           app_title 
        </a>
      </nav>
    </header>
  </appc-header><main
    class="ng-tns-c3-0 ng-trigger ng-trigger-routerTransition"
  >
    <router-outlet
      class="ng-tns-c3-0"
    />
  </main><appc-footer
    class="ng-tns-c3-0"
  >
    <footer
      class="footer text-muted"
    >
      <p
        class="text-muted"
      >
         Â© 2018 app_title 
      </p>
    </footer>
  </appc-footer>
</appc-root>
`;
